### DISCUSSION TITLE
Controlling Zigbee2mqtt Group of Lights with Smart Nora

### DISCUSSION THREAD

**Discussion by Anjerlaan** [andrei-tatar/node-red-contrib-smartnora#235] [Q&A]
Hi Andrei,

I am currently using the Controlicz service (i.c.w. Zigbee2mqtt/Domoticz) to control devices in Google Home.
However this service stops at the end of 2023, so I am investigating Smart Nora as an alternative service, which looks quite promising.
I have installed the free version and I have created some (working) Node-Red flows with Smart Nora scene buttons to control my AV-Devices in Google Home.

The next step is to create flows to control my Zigbee2mqtt Lights.
Most of them I control with Zigbee2mqtt Groups. Now here I get stuck a bit in Node-Red

I managed to create a flow to control a group with 3 dimmable/color_temp GU10 IKEA lights, but it is not exactly working as I want it:

![image](https://user-images.githubusercontent.com/18295963/209309645-dc149790-9e1b-42bb-8f68-c66513487e15.png)

`[{"id":"e97814dc0e728e86","type":"mqtt out","z":"28e4625c5c5f563a","name":"ON/OFF Keuken Licht","topic":"zigbee2mqtt/Keuken_Spots_ct/set","qos":"2","retain":"true","respTopic":"","contentType":"","userProps":"","correl":"","expiry":"","broker":"3f147321.2d0384","x":920,"y":320,"wires":[]},{"id":"5ffa6b78a8c12c88","type":"noraf-light","z":"28e4625c5c5f563a","devicename":"Test Lamp","lightcolor":false,"brightnesscontrol":true,"commandonlycolor":false,"turnonwhenbrightnesschanges":false,"passthru":false,"errorifstateunchaged":false,"statepayload":false,"brightnessoverride":"","roomhint":"Keuken","name":"Test Lamp","colortype":"hsv","nora":"773b9a6c6371d3c5","topic":"zigbee2mqtt/Keuken_Spots_ct/set","onvalue":"true","onvalueType":"bool","offvalue":"false","offvalueType":"bool","temperaturemin":"2700","temperaturemax":"5500","twofactor":"off","twofactorpin":"","filter":false,"x":310,"y":320,"wires":[["a9086e7438eedd8f","2d510d5eee4e3215"]]},{"id":"80e22004026db947","type":"debug","z":"28e4625c5c5f563a","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":870,"y":220,"wires":[]},{"id":"2d510d5eee4e3215","type":"function","z":"28e4625c5c5f563a","name":"toNumber","func":"msg.payload = parseInt(msg.payload,14);\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":480,"y":320,"wires":[["d95f380dae416f41"]]},{"id":"a9086e7438eedd8f","type":"debug","z":"28e4625c5c5f563a","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":510,"y":200,"wires":[]},{"id":"d95f380dae416f41","type":"change","z":"28e4625c5c5f563a","name":"Brightness","rules":[{"t":"set","p":"brightness","pt":"flow","to":"payload","tot":"msg"},{"t":"move","p":"payload","pt":"msg","to":"payload.brightness","tot":"msg"},{"t":"set","p":"payload.on","pt":"msg","to":"true","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":650,"y":320,"wires":[["e97814dc0e728e86","80e22004026db947"]]},{"id":"3f147321.2d0384","type":"mqtt-broker","name":"Zigbee2mqtt","broker":"192.168.1.55","port":"1883","clientid":"","autoConnect":true,"usetls":false,"compatmode":false,"protocolVersion":4,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closeQos":"0","closePayload":"","willTopic":"","willQos":"0","willPayload":""},{"id":"773b9a6c6371d3c5","type":"noraf-config","name":"Google","group":"","twofactor":"off","twofactorpin":"","localexecution":true,"structure":"Anjerlaan","storeStateInContext":false,"disableValidationErrors":false,"sendDeviceNameAndLocation":false}]`

This flow always turns ON the group with brightness level 100 (ON) or 0 (OFF). (I did not take the conversion to 0/255 into account)
However, if I set the brightness level (voice controlled or slider in Google Home) to a certain level and then turn OFF the Group,
That brightness level is remebered. So if I turn ON the Group again, the last brightness level is used.
But I want to control it this way:

If the group is turned ON, the brightness level should always be a default value
The Light in Google Home must be able to controll the brightness level with either the voice or slider.
I would also like to sent a default color_temp.

I have found NORA-Pre and Post Subflows, which I have tested i.c.w. all Light settings, but I cant figure out how to make this work as I want it to work.

Do you maybe have example flows how to set this up in Node-Red, or is it not possible what I want?

Greetzzz,

Gerben

**Comment by andrei-tatar** [comment]
Hi @Anjerlaan,
I guess you could intercept the "on" payload and set the brightness to what you want when the light turns on, just make sure to back feed it into the nora node as well.

Alternatively there is another property you can use:
![image](https://user-images.githubusercontent.com/6473183/209327617-ba8b3e64-b2be-415a-88fd-6642400555db.png)

It basically sets the brightness to a preferred value each time the light turns off.

**Reply by Anjerlaan** [reply]
Hi @andrei-tatar,

How can I intercept ON payload, if "Send/receive complete state as payload" is unchecked?

**Reply by andrei-tatar** [reply]
@Anjerlaan it it's unchecked, you can just this setting (it will always reset the brightness to the set value when you turn off the light):
![image](https://user-images.githubusercontent.com/6473183/210545382-5e648688-f181-45ef-b210-6ad0c687a3b0.png)

**Reply by Anjerlaan** [reply]
@andrei-tatar,

Thanx for the info, but I have a flow that works for me now

`[{"id":"6ee85fdd.12c49","type":"subflow","name":"NORA Sync","info":"","category":"","in":[{"x":80,"y":100,"wires":[{"id":"17537cbfa0caae3f"}]}],"out":[{"x":480,"y":100,"wires":[{"id":"ac59a7b58496fbe5","port":0}]}],"env":[],"meta":{},"color":"#DDAA99"},{"id":"ac59a7b58496fbe5","type":"rbe","z":"6ee85fdd.12c49","name":"","func":"rbe","gap":"","start":"","inout":"out","septopics":true,"property":"payload","topi":"topic","x":370,"y":100,"wires":[[]]},{"id":"17537cbfa0caae3f","type":"function","z":"6ee85fdd.12c49","name":"Sync Google","func":"var check = msg.payload.onvalue;\nvar brightness = Math.round((msg.payload.brightness/255) * 100);\nvar state = msg.payload.state;\n\nif(state == \"ON\" && brightness == check) {\n  \n    msg.payload = {\"brightness\": brightness,\"color\":{\"temperatureK\":2700},\"on\":true,\"online\":true};\n    return msg;\n    \n}\n\nif(state == \"ON\" && brightness !== check && brightness > 0) {\n  \n    msg.payload = {\"brightness\": brightness,\"color\":{\"temperatureK\":2700},\"on\":true,\"online\":true};\n    return msg;\n    \n}\n\nif(state == \"OFF\") {\n  \n    msg.payload = {\"brightness\": 0,\"color\":{\"temperatureK\":2700},\"on\":false,\"online\":true};\n    return msg;\n    \n}\n","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":210,"y":100,"wires":[["ac59a7b58496fbe5"]]},{"id":"d6132d2d.a892c","type":"subflow","name":"NORA Post Light","info":"","category":"","in":[{"x":80,"y":180,"wires":[{"id":"12dc0218bf2a4d8d"}]}],"out":[{"x":920,"y":180,"wires":[{"id":"abc05bedf3d05a2b","port":0},{"id":"4dd1e69a41b04202","port":0},{"id":"783dd8df797fb9c2","port":0}]}],"env":[],"meta":{},"color":"#DDAA99"},{"id":"12dc0218bf2a4d8d","type":"switch","z":"d6132d2d.a892c","name":"On/Off","property":"payload.on","propertyType":"msg","rules":[{"t":"true"},{"t":"false"}],"checkall":"true","repair":false,"outputs":2,"x":210,"y":180,"wires":[["76d77ee6e0df77fc"],["783dd8df797fb9c2"]]},{"id":"783dd8df797fb9c2","type":"change","z":"d6132d2d.a892c","name":"Off","rules":[{"t":"set","p":"#:(file)::onPrevious","pt":"flow","to":"payload.on","tot":"msg"},{"t":"set","p":"payload","pt":"msg","to":"0","tot":"num"}],"action":"","property":"","from":"","to":"","reg":false,"x":590,"y":240,"wires":[[]]},{"id":"76d77ee6e0df77fc","type":"switch","z":"d6132d2d.a892c","name":"onPrevious","property":"#:(file)::onPrevious","propertyType":"flow","rules":[{"t":"true"},{"t":"false"}],"checkall":"true","repair":false,"outputs":2,"x":410,"y":140,"wires":[["f48503e844dc6888"],["4dd1e69a41b04202"]]},{"id":"4dd1e69a41b04202","type":"change","z":"d6132d2d.a892c","name":"Last","rules":[{"t":"set","p":"#:(file)::onPrevious","pt":"flow","to":"payload.on","tot":"msg"},{"t":"set","p":"payload","pt":"msg","to":"255","tot":"num"}],"action":"","property":"","from":"","to":"","reg":false,"x":590,"y":180,"wires":[[]]},{"id":"abc05bedf3d05a2b","type":"function","z":"d6132d2d.a892c","name":"toNumber","func":"/* var brightness = parseInt(msg.payload.brightness,16); */\nvar brightness = Math.round(msg.payload.brightness * 2.55);\n\nmsg.payload.brightness = brightness;\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":800,"y":100,"wires":[[]]},{"id":"f48503e844dc6888","type":"change","z":"d6132d2d.a892c","name":"Brightness","rules":[{"t":"set","p":"#:(file)::onPrevious","pt":"flow","to":"payload.on","tot":"msg"},{"t":"set","p":"payload.brightness","pt":"msg","to":"payload.brightness","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":610,"y":100,"wires":[["abc05bedf3d05a2b"]]},{"id":"e97814dc0e728e86","type":"mqtt out","z":"28e4625c5c5f563a","name":"Keuken Licht","topic":"zigbee2mqtt/Keuken_Spots_ct/set","qos":"2","retain":"true","respTopic":"","contentType":"","userProps":"","correl":"","expiry":"","broker":"3f147321.2d0384","x":1570,"y":140,"wires":[]},{"id":"5ffa6b78a8c12c88","type":"noraf-light","z":"28e4625c5c5f563a","devicename":"Keuken Licht","lightcolor":true,"brightnesscontrol":true,"commandonlycolor":false,"turnonwhenbrightnesschanges":false,"passthru":false,"errorifstateunchaged":false,"statepayload":true,"brightnessoverride":"","roomhint":"Keuken","name":"Keuken Licht","colortype":"temperature","nora":"773b9a6c6371d3c5","topic":"zigbee2mqtt/Keuken_Spots_ct/set","onvalue":"true","onvalueType":"bool","offvalue":"false","offvalueType":"bool","temperaturemin":"2700","temperaturemax":"5500","twofactor":"off","twofactorpin":"","filter":false,"x":970,"y":140,"wires":[["3812005356693103"]]},{"id":"3812005356693103","type":"subflow:d6132d2d.a892c","z":"28e4625c5c5f563a","name":"","x":1170,"y":140,"wires":[["26fb6c52a0bc9961"]]},{"id":"2ea9257fd9e5157e","type":"debug","z":"28e4625c5c5f563a","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":1570,"y":200,"wires":[]},{"id":"4e838f9848e8cb14","type":"mqtt in","z":"28e4625c5c5f563a","name":"Keuken Licht","topic":"zigbee2mqtt/Keuken_Spots_ct","qos":"2","datatype":"auto","broker":"3f147321.2d0384","nl":false,"rap":false,"inputs":0,"x":90,"y":140,"wires":[["2543401354812a37"]]},{"id":"32f956ba2399a433","type":"json","z":"28e4625c5c5f563a","name":"","property":"payload","action":"","pretty":false,"x":390,"y":140,"wires":[["a01ea2f6a0f98c1e","5e299453189d9658"]]},{"id":"26fb6c52a0bc9961","type":"function","z":"28e4625c5c5f563a","name":"On/Off/Dim","func":"if(msg.payload === 255) {\n  \n    msg.payload = {\"state\": \"ON\", \"brightness\":140, \"color_temp\": 350};\n    \n}\n\nif(msg.payload === 0) {\n  \n    msg.payload = {\"state\": \"OFF\", \"brightness\":0, \"color_temp\": 350};\n    \n}\n\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":1370,"y":140,"wires":[["e97814dc0e728e86","2ea9257fd9e5157e"]]},{"id":"a01ea2f6a0f98c1e","type":"change","z":"28e4625c5c5f563a","name":"Set ON Value [%]","rules":[{"t":"set","p":"payload.onvalue","pt":"msg","to":"55","tot":"num"}],"action":"","property":"","from":"","to":"","reg":false,"x":570,"y":140,"wires":[["243b6f3b153b17ab"]]},{"id":"58546057e199c4c8","type":"debug","z":"28e4625c5c5f563a","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":970,"y":200,"wires":[]},{"id":"243b6f3b153b17ab","type":"subflow:6ee85fdd.12c49","z":"28e4625c5c5f563a","name":"","x":790,"y":140,"wires":[["5ffa6b78a8c12c88","58546057e199c4c8"]]},{"id":"caa859f88d866b70","type":"comment","z":"28e4625c5c5f563a","name":"Keuken Licht Flow","info":"","x":930,"y":40,"wires":[]},{"id":"5e299453189d9658","type":"change","z":"28e4625c5c5f563a","name":"Save State to File","rules":[{"t":"move","p":"payload.state","pt":"msg","to":"#:(file)::keuken_remote","tot":"flow"}],"action":"","property":"","from":"","to":"","reg":false,"x":570,"y":80,"wires":[[]]},{"id":"2543401354812a37","type":"delay","z":"28e4625c5c5f563a","name":"Delay","pauseType":"delay","timeout":"10","timeoutUnits":"milliseconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":250,"y":140,"wires":[["32f956ba2399a433"]]},{"id":"3f147321.2d0384","type":"mqtt-broker","name":"Zigbee2mqtt","broker":"192.168.1.55","port":"1883","clientid":"","autoConnect":true,"usetls":false,"compatmode":false,"protocolVersion":4,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closeQos":"0","closePayload":"","willTopic":"","willQos":"0","willPayload":""},{"id":"773b9a6c6371d3c5","type":"noraf-config","name":"Google","group":"","twofactor":"off","twofactorpin":"","localexecution":true,"structure":"Anjerlaan","storeStateInContext":false,"disableValidationErrors":false,"sendDeviceNameAndLocation":false}]`
